loopin.go

```diff
@@ -500,9 +498,9 @@ func (s *loopInSwap) execute(mainCtx context.Context,
 	subCtx, cancel := context.WithCancel(mainCtx)
 	defer cancel()
 
-	s.wg.Add(1)
+	wg.Add(1)
 	go func() {
-		defer s.wg.Done()
+		defer wg.Done()
```

Comment:
So below we use the `subCtx` which will be canceled as soon as the invoice is finalized. I think instead we'd want to use a different context. The `mainCtx` that we pass in to this function lives the whole time the executor is alive so perhaps we need to make a cancellable context per swap that actually cancels when the swap execution is really finished.

Reply:
When we pass `subCtx` to `executeSwap` and cancel the context when the swap ends it works for all cases except for when we resume swaps: https://github.com/lightninglabs/loop/blob/master/client.go#L373
Here we pass the executor context, I haven't figured out yet why that deadlocks the swap execution.